aten/src/ATen/native/BlasKernel.cpp

```diff
@@ -356,4 +358,84 @@ INSTANTIATE_VDOT_IMPL(c10::complex<double>);
 
 #undef INSTANTIATE_DOT_IMPL
 
+namespace blas_impl {
+
+template <typename scalar_t>
+void addr_naive(int64_t m, int64_t n, scalar_t alpha, scalar_t *x, int64_t incx, scalar_t *y, int64_t incy, scalar_t *a, int64_t lda) {
+  for (int64_t j = 0; j < n; j++) {
+    scalar_t* column_ = a + j * lda;
+    scalar_t z = alpha * y[j * incy];
+    for (int64_t i = 0; i < m; i++)
+      column_[i] += z * x[i * incx];
+  }
+}
+
+template <typename scalar_t>
+bool addr_use_fast_path(int64_t m, int64_t n, int64_t lda, int64_t incx, int64_t incy) {
+  return false;
+}
+
+template <typename scalar_t>
+void addr_fast_path(int m, int n, scalar_t alpha, scalar_t *x, int incx, scalar_t *y, int incy, scalar_t *a, int lda) {
+  TORCH_INTERNAL_ASSERT(
+    false,
+    "addr_fast_path shouldn't be called for this configuration"
+  );
+}
+
+#if AT_BUILD_WITH_BLAS()
+
+template <>
+bool addr_use_fast_path<float>(int64_t m, int64_t n, int64_t lda, int64_t incx, int64_t incy) {
+  return (m <= INT_MAX) && (n <= INT_MAX) && (lda <= INT_MAX) &&
+         (incx > 0) && (incx <= INT_MAX) && (incy > 0) && (incy <= INT_MAX);
+}
+
+template <>
+bool addr_use_fast_path<double>(int64_t m, int64_t n, int64_t lda, int64_t incx, int64_t incy) {
+  return addr_use_fast_path<float>(m, n, lda, incx, incy);
+}
+
+template <>
+void addr_fast_path(int m, int n, float alpha, float *x, int incx, float *y, int incy, float *a, int lda) {
+  sger_(&m, &n, &alpha, x, &incx, y, &incy, a, &lda);
+}
+
+template <>
+void addr_fast_path(int m, int n, double alpha, double *x, int incx, double *y, int incy, double *a, int lda) {
+  dger_(&m, &n, &alpha, x, &incx, y, &incy, a, &lda);
+}
+#endif
+}
+
+template<typename scalar_t>
+void addr_impl(int64_t m, int64_t n, scalar_t alpha, scalar_t *x, int64_t incx, scalar_t *y, int64_t incy, scalar_t *a, int64_t lda) {
+  if(n == 1) {
```

Comment:
`if(` -> `if (`

Reply:
Will address in the later commit of this PR.